src/vs/workbench/services/policies/common/multiplexPolicyService.ts

```diff
@@ -58,4 +58,9 @@ export class MultiplexPolicyService extends AbstractPolicyService implements IPo
 			changed.add(key);
 		}
 	}
+
+	public override dispose(): void {
+		this.policyServices.forEach(service => service.dispose());
+		super.dispose();
+	}
```

Comment:
This addition to `multiplexPolicyService` resulted in many of the changes in this PR (ie: updating the interface to extend `IDisposable`, ensuring we `register(...)` `NullPolicyService`, etc...)

Reply:
Upon more discussion this class should not be in charge of disposing it's composed services.